packages/e2e-test-utils-playwright/src/editor/open-document-settings-sidebar.ts

```diff
@@ -5,14 +5,18 @@ import type { Editor } from './index';
 import { expect } from '../test';
 
 /**
- * Clicks on the button in the header which opens Document Settings sidebar when it is closed.
+ * Clicks on the button in the header which opens Document Settings sidebar when
+ * it is closed.
  *
  * @param {Editor} this
  */
 export async function openDocumentSettingsSidebar( this: Editor ) {
-	const editorSettingsButton = this.page.locator(
-		'role=region[name="Editor top bar"i] >> role=button[name="Settings"i]'
-	);
+	const editorSettingsButton = this.page
+		.getByRole( 'region', { name: 'Editor top bar' } )
+		.getByRole( 'button', {
+			name: 'Settings',
+			disabled: false,
+		} );
```

Comment:
Is it preferred to write locators this way, with chained methods instead of the Playwright selector language, or whatever that `role=...` string is? (I couldn't find any docs on this syntax earlier today when I tried.)

Reply:
I'd say that it's the preferred way according to Playwright, as it's what is suggested when using their [built-in test generator](https://playwright.dev/docs/getting-started-vscode#generating-tests). It's more readable, flexible, and robust compared to text selectors. In this specific case, for example, the `disabled` value is automatically derived from either `disabled` or `aria-disabled` attribute, which would need to be explicitly requested via a classic CSS selector.

Reply:
What confuses me, however, is that the `role=region[name="x"]` string is not a CSS or XPath selector, but something else, Playwright-specific. I didn't find docs for it, but it maps 1:1 to the `getRole( 'region', { name: 'x' } )` syntax.

Reply:
It's probably because they're not recommending those selectors anymore, so they've removed it from the documentation. 🤷 I can see a lot of `not recommended` stuff in the [Other selectors](https://playwright.dev/docs/other-locators) section now, which seems to have been updated with the [recent releases](https://playwright.dev/docs/release-notes) (we're currently using v1.27). I can also see that locator chaining is now [the recommended way](https://playwright.dev/docs/locators#chaining-locators). That documentation changes very often, and I can't see a way of going back, at least a version or two, which has also been a pain point for me. 😕

Reply:
Seems like they removed the doc for the text-based `role=` selector 😞 . It should be a one-to-one mapping to the now recommended `getByRole` though. We probably would want to migrate the existing usages when we upgrade playwright, and update our [doc](https://github.com/WordPress/gutenberg/blob/trunk/docs/contributors/code/e2e/README.md#use-accessible-selectors).

Reply:
It's bad that they completely removed the docs for `page.locator`. The text-based locators are more flexible in some cases. Should we file an issue to restore them, at least on some level?

Reply:
We can try that but I think they reached a decision already 😅 . I prefer text-based role selectors in most cases too, but I also don't mind using `getByRole` if that's what is officially recommended.

Reply:
> We can try that but I think they reached a decision already

If they want to deprecate the text-based `role=` selectors, that's fine, no problem with that decision,  but removing the documentation is a strictly bad idea. Existing codebases still use them and they still work, so they should be documented.